src/mtcp/mtcp_restart.c

```diff
@@ -692,6 +692,23 @@ compute_vdso_vvar_addr(RestoreInfo *rinfo)
   mtcp_sys_close(mapsfd);
 }
 
+NO_OPTIMIZE
+static void
+unmap_one_memory_area(Area *area, int mapsfd)
+{
+  int mtcp_sys_errno;
+  DPRINTF("***INFO: munmapping (%p..%p)\n", area->addr, area->endAddr);
+  if (mtcp_sys_munmap(area->addr, area->size) == -1) {
+    MTCP_PRINTF("***WARNING: %s(%x): munmap(%p, %d) failed; errno: %d\n",
+                area->name, area->flags, area->addr, area->size,
+                mtcp_sys_errno);
+    mtcp_abort();
+  }
+
+  // Rewind and reread maps.
```

Comment:
I know that the comment `Rewind and reread maps.` was there earlier.

But let's take this opportunity to document our design decisions.  Normally, we just view the memory regions in order, and unmap them.  Why do we unmap and then start reading again from the beginning?
Perhaps you can give a _brief_ example where this is needed.